### PR TITLE
Fixes #2581

### LINKED ISSUE #2581: egpObjectContainsPoint does not work properly with parents
**Describe the bug**
I was working on a ui library that makes heavy use of parenting and stumbeld upon this bug while trying to add buttons.
If you parent something and then try egpObjectContainsPoint on the child the position will be offset from where it should be. Sorry if im saying giberish ive been awake for way to long working on this library.

**To Reproduce**
```
@inputs EGP:wirelink

EGP:egpClear()

EGP:egpBox(1, vec2(256), vec2(300))

EGP:egpBox(2, vec2(0), vec2(50))
EGP:egpColor(2, vec(255, 0, 0))
EGP:egpParent(2, 1)

print(EGP:egpObjectContainsPoint(2, vec2(256)) ? "does contain" : "does not contain")

```
output: does not contain
image

If i try the exact same thing but without parenting it works flawlessly

```
@inputs EGP:wirelink

EGP:egpClear()

EGP:egpBox(1, vec2(256), vec2(300))

EGP:egpBox(2, vec2(256), vec2(50))
EGP:egpColor(2, vec(255, 0, 0))

print(EGP:egpObjectContainsPoint(2, vec2(256)) ? "does contain" : "does not contain")
```

### PR DESCRIPTION
**Pull request by Denneisk**
Fixes #2581. Tested against a few modified versions of examples.
Also fixes issues with EGPCursor and EGP Emitter rendering.

### CODE REVIEW

**Review by Meowlan — APPROVED** [review]
I want to fucking marry you.

File: lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua [hunk 1]
```diff
@@ -444,8 +444,6 @@ function EGP:EGPCursor( this, ply )
 end
 
 function EGP.ScreenSpaceToObjectSpace(object, point)
-	point = { x = point.x - object.x, y = point.y - object.y }
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua:447]: Won't this break other things? Translation has to also be accounted for, right?

Comment (Denneisk) [reply lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua:447]: This function isn't called anywhere else but in `Contains` (which itself isn't called anywhere but in `egpObjectContainsPoint`). This avoids having to pass the EGP entity to the function in order to reorient itself to the global position using `GetGlobalPos`. An equivalent statement is done in `egpObjectContainsPoint` now.

Comment (thegrb93) [reply lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua:447]: If it's not used anywhere else, can we just get rid of the functions? They aren't doing what their names imply now after your change.

Comment (Denneisk) [reply lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua:447]: Works for me. Slipped in some other fixes, shouldn't be too disastrous.

**Comment by Meowlan** [comment]
I think an e2 function to get the base position of an element would also be useful. My current solution is fairly silly:
```
function vector2 getBasePos(Index) {
    local Parent = EGP:egpParent(Index)
    local Pos = EGP:egpPos(Index)
    while (Parent) {
        Pos = Pos + EGP:egpPos(Parent)
        Parent = EGP:egpParent(Parent)
    }
    
    return Pos
}
```

**Comment by Denneisk** [comment]
> I think an e2 function to get the base position of an element would also be useful. My current solution is fairly silly:
That's called `wirelink:egpGlobalPos()`

**Comment by Meowlan** [comment]
i am quite silly

**Pull request merged**